presto-main/src/main/java/com/facebook/presto/operator/scalar/UnitFunctions.java

```diff
@@ -0,0 +1,175 @@
+    {
+        Duration duration;
+        try {
+            duration = Duration.succinctDuration(value, valueOfTimeUnit(unit.toStringUtf8()));
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, NaN, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+        return utf8Slice(duration.toString());
+    }
+
+    @Description("Returns the succinct string representation of a nanosecond value")
+    @ScalarFunction("succinct_nanos")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctNanos(@SqlType(StandardTypes.BIGINT) long value)
+    {
+        return succinctNanos((double) value);
+    }
+
+    @Description("Returns the succinct string representation of a nanosecond value")
+    @ScalarFunction("succinct_nanos")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctNanos(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        Duration duration;
+        try {
+            duration = Duration.succinctDuration(value, NANOSECONDS);
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+        return utf8Slice(duration.toString());
+    }
+
+    @Description("Returns the succinct string representation of a millisecond value")
+    @ScalarFunction("succinct_millis")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctMillis(@SqlType(StandardTypes.BIGINT) long value)
+    {
+        return succinctMillis((double) value);
+    }
+
+    @Description("Returns the succinct string representation of a millisecond value")
+    @ScalarFunction("succinct_millis")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctMillis(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        Duration duration;
+        try {
+            duration = Duration.succinctDuration(value, MILLISECONDS);
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+        return utf8Slice(duration.toString());
+    }
+
+    @Description("Returns the succinct string representation of a data size")
+    @ScalarFunction("succinct_data_size")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice succinctDataSize(@SqlType(StandardTypes.DOUBLE) double value, @SqlType(StandardTypes.VARCHAR) Slice unit)
+    {
+        DataSize dataSize;
+        try {
+            dataSize = DataSize.succinctDataSize(value, valueOfUnit(unit.toStringUtf8()));
+        }
+        catch (IllegalArgumentException e) {
+            // When value is negative, NaN, etc.
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+        return utf8Slice(dataSize.toString());
+    }
+
+    @Description("Returns the succinct string representation of a byte size")
```

Comment:
`byte size` sounds weird. 

One alternative `"Returns the succinct string representation of a data size specified in bytes"`.